packages/mdx/test/compile.js

```diff
@@ -504,6 +504,38 @@ test('compile', async () => {
     )
   }
 
+  try {
+    renderToStaticMarkup(
+      React.createElement(
+        await run(compileSync('export const a = {}\n\n<a.b />'))
+      )
+    )
+    assert.unreachable()
+  } catch (/** @type {unknown} */ error) {
+    const exception = /** @type {Error} */ (error)
+    assert.match(
+      exception.message,
+      /Expected component `a.b` to be defined/,
+      'should throw if a required member is not passed'
+    )
+  }
+
+  try {
+    renderToStaticMarkup(
+      React.createElement(
+        await run(compileSync('<a render={(x) => <x.y />} />'))
+      )
+    )
+    assert.unreachable()
+  } catch (/** @type {unknown} */ error) {
+    const exception = /** @type {Error} */ (error)
+    assert.match(
+      exception.message,
+      /x is not defined/,
```

Comment:
Wait, why is this error thrown here? Is `render` being called?

Reply:
It does. That's exactly the issue I fix in #1986. This line is added:

```js
if (!x) _missingMdxReference("x", false);',
```

But `x` is not there, it's defined inside an arrow function.